test/Promise-test.js

```diff
@@ -14,32 +14,58 @@ describe('Promise', () => {
 	})
 
 	it('should reject if resolver throws synchronously', () => {
-		let expected = new Error()
+		const expected = new Error()
 		return new Promise(() => { throw expected })
 			.then(assert.ifError, x => assert.strictEqual(expected, x))
 	})
 
-	it('should fulfill with value', () => {
-		let expected = {}
-		return new Promise(resolve => resolve(expected))
-			.then(x => assert.strictEqual(expected, x))
-	})
+	describe('resolvers', () => {
+		it('should fulfill with value', () => {
+			const expected = {}
+			return new Promise(resolve => resolve(expected))
+				.then(x => assert.strictEqual(expected, x))
+		})
 
-	it('should resolve to fulfilled promise', () => {
-		let expected = {}
-		return new Promise(resolve => resolve(fulfill(expected)))
-			.then(x => assert.strictEqual(expected, x))
-	})
+		it('should resolve to fulfilled promise', () => {
+			const expected = {}
+			return new Promise(resolve => resolve(fulfill(expected)))
+				.then(x => assert.strictEqual(expected, x))
+		})
 
-	it('should resolve to rejected promise', () => {
-		let expected = {}
-		return new Promise(resolve => resolve(reject(expected)))
-			.then(assert.ifError, x => assert.strictEqual(expected, x))
-	})
+		it('should resolve to rejected promise', () => {
+			const expected = new Error()
+			return new Promise(resolve => resolve(reject(expected)))
+				.then(assert.ifError, x => assert.strictEqual(expected, x))
+		})
 
-	it('should reject with value', () => {
-		let expected = {}
-		return new Promise((resolve, reject) => reject(expected))
-			.then(assert.ifError, x => assert.strictEqual(expected, x))
+		it('should reject with value', () => {
+			const expected = new Error()
+			return new Promise((resolve, reject) => reject(expected))
+				.then(assert.ifError, x => assert.strictEqual(expected, x))
+		})
+
+		it('should asynchronously fulfill with value', () => {
```

Comment:
Hmmm, can you explain the value in these async variants of the tests above?  is it to verify that there's no functional difference between calling resolve/reject before the resolver function returns vs after it returns?